### PR TITLE
[PUBDEV-4252] Allow to set timeout for external write confirmation

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

**Comment by mmalohlava** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/ExternalFrameWriterClient.java [hunk 1]
```diff
@@ -142,13 +143,36 @@ public void sendNA() throws IOException{
...
+            // confirmAb.getInt() forces this code to wait for result and
+            // forces all the previous work to be done on the recipient side.
+            Callable<Byte> task = new Callable<Byte>() {
+                public Byte call() {
+                    return confirmAb.get1();
+                }
+            };
+
+            Future<Byte> future = Executors.newFixedThreadPool(1).submit(task);
+            try {
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/ExternalFrameWriterClient.java]: Why not to use FJ? you do not need executor service...

File: h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java [hunk 2]
```diff
@@ -88,7 +88,11 @@ public void run() {
                         writer.sendNA();
                         writer.sendNA();
 
-                        writer.waitUntilAllWritten();
+                        try {
+                            writer.waitUntilAllWritten(10);
+                        }catch (ExternalFrameConfirmationException e){
+                            e.printStackTrace();
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java]: no assertion in test?

Comment (jakubhava) [inline h2o-core/src/test/java/water/ExternalFrameWriterClientTest.java]: The assertion is already inside the `waitUntilAllWritten` method

**Comment by jakubhava** [comment]
@h2o-ops please test!

**Pull request merged**